### ISSUE TITLE
UIDevice.name neutered in iOS 16

### ISSUE THREAD

**Issue by gutley** [mozilla-mobile/firefox-ios#10943]
From the headers:

``` swift
    open var name: String { get } // Synonym for model. Prior to iOS 16, user-assigned device name (e.g. @"My iPhone").
```

I can see its used in a few places throughout the code, so a replacement may be needed.

From the "What's new in UIKit" talk at WWDC22 it mentions that you can get the old behaviour via a new entitlement, but I haven't yet tracked down what that entitlement is and how to get it.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-4349)

**Comment by przemyslaw-szurmak** [comment]
@gutley inside https://developer.apple.com/wwdc22/10096 they explained app needs to get `com.apple.developer.device-information.user-assigned-device-name` but I couldn't find any information on how to get it...

**Comment by lmarceau** [comment]
We also have occurances of UIDevice.orientation in our code, so we should replace those too.

**Comment by gutley** [comment]
FYI - Documentation on how to request the correct entitlement is now available:
https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_developer_device-information_user-assigned-device-name

**Comment by data-sync-user** [comment]
➤ Orla Mitchell commented:

We requested the entitlement from apple several weeks ago, prior to the launch of iOS 16, but they still haven’t approved us.

In the mean time since iOS 16 was launched this change does not seems to have had any adverse effects on the app. I would have though the downstream impact of this would be that any synced tabs coming from an iOS device would show up under a new name when viewed from desktop or any other device (ex: iPhone when it would have previously been Orla’s iPhone). This does not seem to be the case tho when I did some quick testing around this issue, so I think we can close this ticket. 

Also since the upgrade has already happened, even if there were issues that I haven’t been able to observe I think it’s too late to change them now as it will just cause extra disruption. 

Daniela Arcese thoughts?
